hack/lib/etcd.sh

```diff
@@ -27,9 +27,9 @@ kube::etcd::validate() {
     exit 1
   }
 
-  # validate it is not running
-  if pgrep -x etcd >/dev/null 2>&1; then
-    kube::log::usage "etcd appears to already be running on this machine (`pgrep -xl etcd`) (or its a zombie and you need to kill its parent)."
+  # validate etcd port is free
+  if netstat -nlt | grep ':'$ETCD_PORT' ' >/dev/null 2>&1; then
```

Comment:
I was testing this on my darwin system, and unfortunately the flags I provided didn't work.

However, something like this did:
```
netstat -nat | grep "[\.:]${ETCD_PORT:?} .*LISTEN"
```

Can you test and see if that works for you?

Reply:
works! done.